File: app/src/main/java/jgappsandgames/smartreminderslite/tags/TagActivityInterface.java
package jgappsandgames.smartreminderslite.tags;

import android.app.Activity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.holder.TagHolder;
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder;

/**
 * TagActivityInterface
 * Created by joshua on 11/29/2017.
 */
public abstract class TagActivityInterface extends Activity implements TagHolder.TagSwitcher, TaskFolderHolder.OnTaskChangedListener {
    // Views
    protected TextView tasks_text;
    protected ListView tasks_list;
    protected TextView selected_text;
    protected ListView selected_list;
    protected TextView unselected_text;
    protected ListView unselected_list;

    // Adapters
    protected TaskAdapter task_adapter;
    protected SelectedAdapter selected_adapter;
    protected UnselectedAdapter unselected_adapter;

    // LifeCycle Methods
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Set View
        setContentView(R.layout.activity_tag);


        // Find Views
        tasks_text = findViewById(R.id.tasks_title);
        tasks_list = findViewById(R.id.tasks);
        selected_text = findViewById(R.id.selected_text);
        selected_list = findViewById(R.id.selected);
        unselected_text = findViewById(R.id.unselected_text);
        unselected_list = findViewById(R.id.unselected);
    }

    // Menu
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_auxilary, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.save:
                save();
                Toast.makeText(this, "Saved.", Toast.LENGTH_LONG).show();
                break;

            case R.id.close:
                finish();
        }

        return super.onOptionsItemSelected(item);
    }

    // Abstract Methods
    public abstract void save();
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/tags/TagActivityInterface.kt
package jgappsandgames.smartreminderslite.sort.tags

// Activity
import android.app.Activity
import android.os.Bundle

// Views
import android.view.Menu
import android.view.MenuItem
import android.widget.ListView
import android.widget.TextView
import android.widget.Toast

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.adapter.TagAdapterInterface
import jgappsandgames.smartreminderslite.holder.TagHolder
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder

/**
 * TagActivityInterface
 * Created by joshua on 1/19/2018.
 */
abstract class TagActivityInterface: Activity(), TagHolder.TagSwitcher, TaskFolderHolder.OnTaskChangedListener {
    // Views ---------------------------------------------------------------------------------------
    private var tasksText: TextView? = null
    protected var tasksList: ListView? = null
    private var selectedText: TextView? = null
    protected var selectedList: ListView? = null
    private var unselectedText: TextView? = null
    protected var unselectedList: ListView? = null

    // Adapters ------------------------------------------------------------------------------------
    protected var taskAdapter: TaskAdapter? = null
    protected var selectedAdapter: TagAdapterInterface? = null
    protected var unselectedAdapter: TagAdapterInterface? = null

    // LifeCycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tag)

        // Find Views
        tasksText = findViewById(R.id.tasks_title)
        tasksList = findViewById(R.id.tasks)
        selectedText = findViewById(R.id.selected_text)
        selectedList = findViewById(R.id.selected)
        unselectedText = findViewById(R.id.unselected_text)
        unselectedList = findViewById(R.id.unselected)
    }

    // Menu ----------------------------------------------------------------------------------------
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_auxilary, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.save -> {
                save()
                Toast.makeText(this, "Saved.", Toast.LENGTH_LONG).show()
            }

            R.id.close -> finish()
        }

        return super.onOptionsItemSelected(item)
    }

    // Abstract Methods ----------------------------------------------------------------------------
    abstract fun save()
}